src/devices/cpu/gigatron/gigatron.cpp

```diff
@@ -0,0 +1,129 @@
+// license:BSD-3-Clause
+// copyright-holders:<author_name>
+/*****************************************************************************
+ *
+ * Skeleton device for Gigatron CPU Core
+ *
+ *****************************************************************************/
+
+#include "emu.h"
+#include "gigatron.h"
+#include "debugger.h"
+
+
+DEFINE_DEVICE_TYPE(GTRON, gigatron_cpu_device, "gigatron", "Gigatron CPU Device")
+
+
+/* FLAGS */
+#if 0
+#define S  0x80
+#define Z  0x40
+#define OV 0x20
+#define C  0x10
+#endif
+
+
+#define gigatron_readop(A) m_program->read_dword(A)
+#define gigatron_readmem16(A) m_data->read_dword(A)
+#define gigatron_writemem16(A,B) m_data->write_dword((A),B)
+
+
+/***********************************
+ *  illegal opcodes
+ ***********************************/
+void gigatron_cpu_device::gigatron_illegal()
+{
+	logerror("gigatron illegal opcode at 0x%04x\n", m_pc);
+	m_icount -= 1;
+}
+
+/* Execute cycles */
+void gigatron_cpu_device::execute_run()
+{
+	uint16_t opcode;
+
+	do
+	{
+		debugger_instruction_hook(this, m_pc);
+
+		opcode = gigatron_readop(m_pc);
+		m_pc++;
+
+		switch( opcode )
+		{
+			default:
+				gigatron_illegal();
+				break;
+		}
+
+	} while( m_icount > 0 );
+}
+
+
+void gigatron_cpu_device::device_start()
+{
+	m_program = &space(AS_PROGRAM);
+	m_data = &space(AS_DATA);
+
+	save_item(NAME(m_pc));
+	save_item(NAME(m_flags));
+
+	// Register state for debugger
+	state_add( GTRON_R0, "PC", m_pc ).formatstr("%02X");
+	state_add( STATE_GENPC, "GENPC", m_r[7] ).noshow();
+	state_add( STATE_GENPCBASE, "CURPC", m_r[7] ).noshow();
+	state_add( STATE_GENFLAGS, "GENFLAGS", m_flags ).noshow();
+
+	m_icountptr = &m_icount;
+}
+
+#if 0
+void gigatron_cpu_device::execute_set_input(int irqline, int state)
+{
+	switch(irqline)
```

Comment:
Please be consistent in your spacing.  At a glance, I can see
* `while( m_icount > 0 )` (no space between keyword and condition, spaces between parentheses and expression),
* `switch(irqline)` (no spaces at all), and
* `if (CLEAR_LINE != state)` (space between keyword and condition, no space between parentheses and condition).

Please choose a style and stick with it.  Most of MAME uses the third style, so those would normalise as:
* `while (m_icount > 0)`
* `switch (irqline)`
* `if (CLEAR_LINE != state)`